pkg/server/utils.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
What do you think if we move `utils.go` into `pkg/server/testing/utils.go` or into `pkg/testing/utils.go`? I think it would clearly state that the utils are applicable only for test cases. In addition, they would have a nice usage scenario, like
```
go testing.SpdkMockServer(ln, tt.spdk)
```

Reply:
yeah, frankly I don't like `server` here, so I intend to rename it to maybe `utils` in a separate PR... then `utils.something` maybe also looks better...

Reply:
Anyway, I'd put `SpdkMockServer` into `testng` package to have that nice prefix `testing.SpdkMockServer`. If we have `testing` in `utils` or directly in `pkg`, it does not matter, in my opinion 😁

Reply:
let me think about that, since I see other go packages use mockery instead... maybe this is better path forward...

Reply:
If you want to eliminate `SpdkMockServer`, I think we don't need any mock framework. We could just create an interface to perform that spdk call and put an instance of that interface into our `frontend.Server` structure. In the tests, we would just replace that interface with a stub implementation. As a bonus, we would reduce test-flakiness since we eliminate the global state which causes some intermittent test failures.